### DISCUSSION TITLE
Running the queue task in Docker Swarm

### DISCUSSION THREAD

**Discussion by lemoba** [serversideup/docker-php#366] [Q&A]
Thank you for creating such a great product. Currently, I plan to use Docker Swarm to run the Laravel project, but I have a question. Will scrolling through updating queue tasks cause interruptions to running tasks? Here is my docker-compose.yml
![WeChat9e153baa8e506e1572d237246799925d](https://github.com/serversideup/docker-php/assets/31270124/6d671727-e98e-454b-ba1f-d2daf0f549e9)

**Comment by jaydrogers** [comment]
Thank you for your compliments!

My friend @nckrtl and I have been working through some horizon improvements, and the way the Server Side Up docs state how to run the queues -- this will actually cause a job to terminate early.

We have a working solution, but we need to get this solution formalized yet. I am sure Horizon will be very similar to a vanilla queue.

Track this issue for the latest https://github.com/serversideup/docker-php/issues/361

**Comment by nckrtl** [comment]
To add to @jaydrogers, the solution we worked on (prevent running jobs to be interrupted during deployments) focused mainly on running Horizon indeed. But the solution we came up with is rather generic to any running process that can stop gracefully.

However, queue:work is a process that runs in the foreground while horizon runs in the background. Meaning that whenever you stop queue:work you interrupt the running jobs. As far as I know there is no queue:stop variant like there is for horizon (`horizon:terminate`), preventing your queue to be shutdown gracefully (finish running jobs and then exit the process).

So I'd recommend Horizon over the regular queue worker, but you will need access to a Redis instance as it's mandatory for Horizon.

**Reply by jaydrogers** [reply]
Oh, thanks for chiming in on this @nckrtl!

I assumed there was a Queue Stop and put this on my list to investigate further.

Horizon definitely sounds like the better solution in this case and I'd still recommend tracking #361 as we implement the fix that @nckrtl and I worked on 👍
